packages/web/src/common/store/premiumContent/sagas.ts

```diff
@@ -287,8 +293,45 @@ function* updateCollectibleGatedTrackAccess(
   }
 }
 
+const PREMIUM_TRACK_POLL_FREQUENCY = 2000
+
+function* pollPremiumTrack({
+  trackParams,
+  frequency
+}: {
+  trackParams: TrackRouteParams
+  frequency: number
+}) {
+  const { trackId, slug, handle } = trackParams ?? {}
+  while (true) {
+    const premiumTrackSignatureMap = yield* select(getPremiumTrackSignatureMap)
+    if (trackId && premiumTrackSignatureMap[trackId]) {
+      yield* put(updatePremiumTrackStatus({ status: 'UNLOCKED' }))
+      break
```

Comment:
nit: to avoid a `while(true)`/break, could dupe the fetching logic right before the loop, and move this check to the `while` conditional, and then refetch at the end of the loop

Reply:
ah right 👍🏿